src/App.js

```diff
@@ -1,7 +1,89 @@
-import React from "react";
+import { useState } from "react";
 
 const App = () => {
-  return <div>Hello there</div>;
+  const url = "http://localhost:5000";
+  const [task, setTask] = useState({ name: "", id: null });
+
+  const dummyData = { name: "first task", id: 2 };
+  const updatedDummyData = { name: "Updated task", id: 2 };
+
+  const getAllTasks = () => {
+    fetch(url + "/tasks").then(() => console.log(task));
+  };
+
+  const getTask = (id) => {
+    fetch(url + `/tasks/${2}`).then((res) => console.log(res));
+  };
+
+  const postTask = async () => {
+    try {
+      const response = await fetch(url + "/tasks", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(dummyData),
+      });
+      if (response.status === 200) {
+        setTask(dummyData);
+      }
+    } catch (error) {
+      console.log("Error: ", error);
+    }
+  };
+
+  const updateTask = (id) => {
+    fetch(url + `/tasks/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(updatedDummyData),
+    })
+      .then(() => {
+        if (task) {
+          setTask(updatedDummyData);
```

Comment:
You are setting the data to updateDummyData. But the server usually does something to your data. In the PUT and POST endpoint I want you to get the payload from the response and put it in some sort of state.